velox/substrait/SubstraitToVeloxPlan.cpp

```diff
@@ -494,22 +500,27 @@ core::PlanNodePtr SubstraitVeloxPlanConverter::toVeloxPlan(
 core::PlanNodePtr SubstraitVeloxPlanConverter::toVeloxPlan(
     const ::substrait::Plan& substraitPlan,
     memory::MemoryPool* pool) {
-  // Construct the function map based on the Substrait representation.
-  constructFunctionMap(substraitPlan);
+  if (!checkTypeExtension(substraitPlan)) {
+    VELOX_FAIL("Only Support UNKNOWN type in UserDefined type extension");
+  } else {
+    // Construct the function map based on the Substrait representation.
+    constructFunctionMap(substraitPlan);
 
-  // Construct the expression converter.
-  exprConverter_ = std::make_shared<SubstraitVeloxExprConverter>(functionMap_);
+    // Construct the expression converter.
+    exprConverter_ =
+        std::make_shared<SubstraitVeloxExprConverter>(functionMap_);
 
-  // In fact, only one RelRoot or Rel is expected here.
-  for (const auto& rel : substraitPlan.relations()) {
-    if (rel.has_root()) {
-      return toVeloxPlan(rel.root(), pool);
-    }
-    if (rel.has_rel()) {
-      return toVeloxPlan(rel.rel(), pool);
+    // In fact, only one RelRoot or Rel is expected here.
+    for (const auto& rel : substraitPlan.relations()) {
```

Comment:
Given the command above, it would be cleaner to check that substraitPlan.relations() has just one entry. Then check the type of that entry.

Reply:
add check.